File: Bindings~/java/src/main/java/com/alttester/Commands/UnityCommand/AltDeletePlayerPref.java
/*
    Copyright(C) 2026 Altom Consulting
*/

package com.alttester.Commands.UnityCommand;

import com.alttester.AltMessage;
import com.alttester.Commands.AltBaseCommand;
import com.alttester.IMessageHandler;

/** Delete entire player pref of the application. */
public class AltDeletePlayerPref extends AltBaseCommand {
  public AltDeletePlayerPref(IMessageHandler messageHandler) {
    super(messageHandler);
  }

  public void Execute() {
    AltMessage altMessage = new AltMessage();
    altMessage.setCommandName("deletePlayerPref");
    SendCommand(altMessage);
    String data = recvall(altMessage, String.class);
    validateResponse("Ok", data);
  }
}

File: Assets/AltTester/Runtime/AltDriver/Commands/UnityCommands/AltDeletePlayerPref.cs
/*
    Copyright(C) 2026 Altom Consulting
*/

namespace AltTester.AltTesterSDK.Driver.Commands
{
    public class AltDeletePlayerPref : AltBaseCommand
    {
        AltDeletePlayerPrefParams cmdParams;
        public AltDeletePlayerPref(IDriverCommunication commHandler) : base(commHandler)
        {
            this.cmdParams = new AltDeletePlayerPrefParams();
        }
        public void Execute()
        {
            CommHandler.Send(cmdParams);
            var data = CommHandler.Recvall<string>(cmdParams);
            ValidateResponse("Ok", data);
        }
    }
}
